packages/react/src/components/dialog/Dialog.tsx

```diff
@@ -0,0 +1,251 @@
+  | {
+      /**
+       * A function to set isOpen property to `false`. The dialog header will have a close-button if this property exists. The dialog will also close on Escape keypress.
+       */
+      close: () => void;
+      /**
+       * The aria-label for the close button. Required with close property.
+       */
+      closeButtonLabelText: string;
+    }
+  | {
+      close?: undefined;
+      closeButtonLabelText?: undefined;
+    };
+
+export type DialogProps = React.PropsWithChildren<
+  {
+    /**
+     * The id of the dialog element.
+     */
+    id: string;
+    /**
+     * The id of the heading element.
+     */
+    'aria-labelledby': string;
+    /**
+     * The id of the content description element which is usually a paragraph. The paragraph should be part of the content.
+     */
+    'aria-describedby'?: string;
+    /**
+     * When `true`, dialog is visible,
+     */
+    isOpen: boolean;
+    /**
+     * The element which will get focus after the dialog is closed.
+     */
+    focusAfterCloseElement?: HTMLElement;
+    /**
+     * When `true` dialog content is scrollable if needed. Use only for long text contents.
+     */
+    scrollable?: boolean;
+    /**
+     * Defines the dialog variant
+     */
+    variant?: DialogVariant;
+    /**
+     * Additional styles
+     */
+    style?: React.CSSProperties;
+    /**
+     * Custom theme styles
+     */
+    theme?: DialogCustomTheme;
+    /**
+     * Additional class names to apply to the dialog.
+     */
+    className?: string;
+    /**
+     * Target element where dialog is rendered. The dialog is rendered into the document.body by default.
+     */
+    targetElement?: HTMLElement;
+  } & DialogCloseProps
+>;
+
+export const Dialog = ({
+  id,
+  isOpen,
+  children,
+  close,
+  closeButtonLabelText,
+  focusAfterCloseElement,
+  scrollable,
+  variant = 'primary',
+  style,
+  theme,
+  className,
+  targetElement,
+  ...props
+}: DialogProps) => {
+  const dialogContextProps: DialogContextProps = { scrollable, close, closeButtonLabelText };
+  const customThemeClass = useTheme<DialogCustomTheme>(styles.dialogContainer, theme);
+  const dialogRef: RefObject<HTMLInputElement> = React.createRef();
+
+  const { 'aria-labelledby': ariaLabelledby, 'aria-describedby': ariaDescribedby } = props;
+
+  const setFocusAfterClose = useCallback((): void => {
+    if (focusAfterCloseElement) {
+      focusAfterCloseElement.focus();
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn('The "focusAfterCloseElement" property did not match any element.');
+    }
+  }, [focusAfterCloseElement]);
+
+  const onKeyDown = (event: KeyboardEvent): void => {
+    if (close && event.key === 'Escape') {
+      close();
+    }
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      document.body.classList.add(styles.dialogVisibleBody);
+      document.addEventListener('keydown', onKeyDown, false);
+    }
+
+    return (): void => {
+      if (isOpen) {
```

Comment:
Is there a possibility that these won't be removed if modal is closed?

Reply:
Maybe if the dialog element is not closed gracefully (for example, when removing straight from DOM) 🤔